### PR TITLE
Pr 280 support windows

### PR DESCRIPTION
**Pull request by pv**
Rebased https://github.com/spacetelescope/asv/pull/280 plus some additional fixes for Windows.
New PR so that Appveyor tests run.

### CODE REVIEW

**Comment by pv** [comment]
Tests pass currently; discussion on the one additional xfail is in https://github.com/spacetelescope/asv/pull/280#issuecomment-116301165

File: asv/plugins/conda.py [hunk 1]
```diff
@@ -117,11 +121,26 @@ def _install_requirements(self):
...
+        """Find an executable in the environment"""
+        if WIN:
+            executable += ".exe"
+
+            exe = os.path.join(self._path, 'Scripts', executable)
+            if os.path.isfile(exe):
+                return exe
+            exe = os.path.join(self._path, executable)
+            if os.path.isfile(exe):
+                return exe
```
Comment (mdboom) [inline asv/plugins/conda.py]: Maybe move this to util or a base class, since it's repeated in virtualenv.py?

File: asv/util.py [hunk 2]
```diff
@@ -346,75 +357,103 @@ def get_content(header=None):
...
+        watcher.start()
+        try:
+            stdout, stderr = proc.communicate()
+        finally:
+            if proc.returncode is None:
+                proc.terminate()
+                proc.wait()
+            watcher.join()
+
+        is_timeout = was_timeout[0]
```
Comment (mdboom) [inline asv/util.py]: Nice to have timeout support on Windows!

File: test/test_environment.py [hunk 3]
```diff
@@ -13,18 +14,21 @@
...
+WIN = (os.name == "nt")
+
+
 try:
     util.which('python2.7')
     HAS_PYTHON_27 = True
-except RuntimeError:
-    HAS_PYTHON_27 = False
+except (RuntimeError, IOError):
+    HAS_PYTHON_27 = (sys.version_info[:2] == (2, 7))
```
Comment (mdboom) [inline test/test_environment.py]: This logic doesn't seem right.  `HAS_PYTHON_27` is supposed to indicate that python2.7 is available somewhere on the `PATH`, not to indicate what the present version of Python is.

Comment (pv) [inline test/test_environment.py]: Yes, but if the present Python version is 2.7, then 2.7 is available even if `python2.7` is not in PATH. This is the situation when installing Python from Python.org on Windows.

Comment (pv) [inline test/test_environment.py]: (Note that this comes from the new logic in `Virtualenv._find_python`)

Comment (mdboom) [inline test/test_environment.py]: I see.  Sounds good.

**Comment by pv** [comment]
Ok, refactored a bit.

**Comment by pv** [comment]
Ok to merge?

File: asv/environment.py [hunk 4]
```diff
@@ -275,18 +290,18 @@ def create(self):
...
-                    shutil.rmtree(self._path)
+                    util.long_path_rmtree(self._path)
                 raise
 
         self.save_info_file(self._path)
 
         self._is_setup = True
 
-    def _setup(self):
+    def setup(self):
```
Comment (mdboom) [inline asv/environment.py]: Why the change to public?  The original intention here was that this should only be called from `create`...

File: asv/graph.py [hunk 5]
```diff
@@ -118,13 +118,15 @@ def resample_data(self, val):
 
         new_val = []
         j = 0
-        for i in xrange(min_time + step_size, max_time + step_size, step_size):
+        i = min_time + step_size
```
Comment (mdboom) [inline asv/graph.py]: Why this change?  Less likely for a change to inadvertently fail the loop end condition when using `xrange`...

**Comment by mdboom** [comment]
Great to see this working on Windows!

Other than my few minor comments, I think this is ready to merge.  Thanks for this!

File: asv/graph.py (hunk 5 above)
Comment (pv) [inline asv/graph.py]: Javascript timestamps handled here overflow the 32-bit integers on
win32, and xrange on Py2 only works with native size integers, so using
it is not possible here.

File: asv/environment.py (hunk 4 above)
Comment (pv) [inline asv/environment.py]: Ok, reverted `setup` back to `_setup`.

File: asv/graph.py [hunk 6]
```diff
@@ -116,15 +116,21 @@ def resample_data(self, val):
         max_time = max(x[0] for x in val)
         step_size = int((max_time - min_time) / RESAMPLED_POINTS)
 
+        # This loop cannot use xrange, because xrange on Python2 on
+        # 32-bit systems can only deal with 32-bit integers, and
+        # Javascript timestamps (1000*unix_timestamp) handled here
+        # overflow this range
```
Comment (mdboom) [inline asv/graph.py]: Thanks.  The comment addresses this nicely.
